File: src/service-catalog.ts
#!/usr/bin/env node
import "source-map-support/register";
import * as cdk from "@aws-cdk/core";
import { S3BucketProduct } from "./products/s3-bucket";
import { existsSync } from "fs";

export class ServiceCatalogStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // we depend on the synth phase to have been performed
    if (!existsSync("cdk.out")) {
      return;
    }
    new S3BucketProduct(this, "S3BucketProduct");
  }
}

const app = new cdk.App();

if (!process.env.AWS_REGION)
  throw Error("must set environment variable: AWS_REGION");

const { AWS_REGION, CDK_DEFAULT_ACCOUNT } = process.env;
const env = { account: CDK_DEFAULT_ACCOUNT, region: AWS_REGION };

new ServiceCatalogStack(app, "ServiceCatalogStack", {
  env,
});

File: python/src/service_catalog.py
#!/usr/bin/env python3
from os import getenv
from aws_cdk import core
from products.s3_bucket.s3_bucket_product import S3BucketProduct
from pathlib import Path


class ServiceCatalogsStack(core.Stack):

    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)

        if not Path("cdk.out").exists():
            return

        S3BucketProduct(self, "S3BucketProduct")


app = core.App()
if not getenv('AWS_REGION'):
    raise Exception('must set environment variable: AWS_REGION')

_env = core.Environment(account=getenv(
    'CDK_DEFAULT_ACCOUNT'), region=getenv('AWS_REGION'))

ServiceCatalogsStack(app, "ServiceCatalogStack", env=_env)
app.synth()
